uint8 pin
string direction
---
uint8 result
